### DISCUSSION TITLE
Renovate does not read .github/renovate.json on a forked repository

### DISCUSSION THREAD

**Discussion by tksst** [renovatebot/renovate#21020] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I want renovate to update my repository that is the fork of some repository,
And I want to place renovate.json under `.github/`

### What you tried so far.

minimal reproduction repository is [here](https://github.com/tksst/renovate-bug-forked-repository-renovatejson)

The config contains "forkProcessing": "enabled" but renovate says `INFO: Repository is a fork and not manually configured - skipping`

renovate.json is placed at ".github/renovate.json" in this repository and Renovate try to get only "/renovate.json"
But [the doc says renovate.json can be placed under .github/](https://docs.renovatebot.com/configuration-options/)

If I place renovate.json on a top of repository, it works for me. But I want to put it under `.github/`

### Relevant debug logs

<details><summary>Logs</summary>

```
INFO: Repository started
{
  "renovateVersion": "35.10.2"
}
DEBUG: Using localDir: /mnt/renovate/gh/tksst/renovate-bug-forked-repository-renovatejson
DEBUG: PackageFiles.clear() - Package files deleted
DEBUG: initRepo("tksst/renovate-bug-forked-repository-renovatejson")
DEBUG: Using queue: host=api.github.com, concurrency=10
DEBUG: tksst/renovate-bug-forked-repository-renovatejson default branch = main
DEBUG: Using app token for git init
DEBUG: GET https://api.github.com/repos/tksst/renovate-bug-forked-repository-renovatejson/contents/renovate.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=167)
INFO: Repository is a fork and not manually configured - skipping
DEBUG: Repository result: fork, status: disabled, enabled: false, onboarded: undefined
```

</details>

**Comment by rarkins** [comment, marked as answer]
It is an intended limitation that you need to put `forkProcessing` into a `renovate.json` in the root directory of the repository, although not clearly documented enough.

From the docs:

<img width="1004" alt="image" src="https://user-images.githubusercontent.com/6311784/226088028-76a4373d-1816-4228-a829-a9c6187f32b3.png">

The reason for this is that it saves us from cloning every forked repo _just in case_ it has enabled fork processing. Instead, we can use the GitHub API to check just for the `renovate.json` file.

I will try to make this more clear in a future doc update

**Reply by rarkins** [reply]
docs improvement: https://github.com/renovatebot/renovate/pull/21021

**Reply by tksst** [reply]
Thanks for the reply. I understand your explanation.

**Reply by kroese** [reply]
@rarkins Can you please elaborate a bit more why you need the clone the repo? Currently you are using the Github API to check for a file in the root. Wouldnt it be extremely simple to just do that check twice? Once for the root and once for the .github folder? Why would it need to be cloned?

**Reply by rarkins** [reply]
Renovate supports many file names, not just two, and it would be a waste of API calls to try them all

**Reply by kroese** [reply]
@rarkins Then why not create a setting on mend.io where users can choose their preferred location? Then you only need to do 1 API call

**Reply by rarkins** [reply]
We could, but there's many other features we'd prefer to add first which would provide a larger benefit and to more users. Users who need to run Renovate on a fork _and_ who cannot tolerate `renovate.json` is a narrow group compared to most others features. So it's not a bad idea, it's just not a prioritized one
